File: client/discord_bot_manager/lib/botmanager.dart
import 'dart:convert';
import 'dart:ffi';
import 'package:http/http.dart' as http;
class BotManager {
  String serverIp = "http://10.0.0.21:8765/";

  Future<String> startBot(String token) async {
    String responseMessage = "";
    var url = Uri.parse(serverIp + "api/bot/start");
    try {
      var response = await http.post(
        url,
        headers: {"Content-Type": "application/json"},
        body: jsonEncode({"token": token}),
      );
      var jsonResponse = jsonDecode(response.body);
      responseMessage = jsonResponse["message"];
    } catch (e) {
      responseMessage = e.toString();
    }
    return responseMessage;
  }

  Future<String> stopBot(String token) async {
    String responseMessage = "";
    var url = Uri.parse(serverIp + "api/bot/stop");
    try {
      var response = await http.post(
        url,
        headers: {"Content-Type": "application/json"},
        body: jsonEncode({"token": token}),
      );
      var jsonResponse = jsonDecode(response.body);
      responseMessage = jsonResponse["message"];
    } catch (e) {
      responseMessage = e.toString();
    }
    return responseMessage;
  }

  Future<bool> isBotOnline(String token) async {
    var url = Uri.parse(serverIp + "api/bot/isOnline");
    try {
      var response = await http.post(
        url,
        headers: {"Content-Type": "application/json"},
        body: jsonEncode({"token": token}),
      );
      var jsonResponse = jsonDecode(response.body);
      print(jsonResponse);
      return jsonResponse["message"];
    } catch (e) {
      return false;
    }
  }
}

File: backend/DiscordBotManager/src/main/java/dev/tobi/bot/BotManager.java
package dev.tobi.bot;

import java.util.HashMap;
import java.util.Map;

public class BotManager {

    private static final Map<String, DcBot> bots = new HashMap<>();

    public static synchronized DcBot getBotByToken(String token) {
        return bots.get(token);
    }

    public static synchronized void startBot(String token) {
        if (!bots.containsKey(token)) {
            DcBot bot = new DcBot(token);
            bot.start();
            bots.put(token, bot);
        }
        System.out.println(bots);
    }

    public static synchronized void stopBot(String token) {
        DcBot bot = bots.get(token);
        if (bot != null) {
            bot.stop();
            bots.remove(token);
        }
        System.out.println(bots);
    }

    public static synchronized boolean isOnline(String token) {
        return bots.containsKey(token);
    }

    public static Map<String, DcBot> getBots() {
        return bots;
    }
}
